test/index.js

```diff
@@ -33,25 +33,52 @@ function Just(x) {
   return {constructor: MaybeTypeRep, isNothing: false, isJust: true, value: x};
 }
 
+function mock($$type) {
+  return {constructor: {'@@type': $$type}};
+}
+
+function parsed(namespace, name, version) {
+  return {namespace: namespace, name: name, version: version};
+}
+
+function named(name) {
+  return parsed(null, name, null);
+}
```

Comment:
I added the above three helper functions to allow every assertion to fit on one line, and keep it readable.